### PR TITLE
Make Permissions partially-ordered.

### PR DESCRIPTION
**Pull request by mmirate**
Specifically:

* `P1 <= P2` iff `P1` expresses a subset of the permissions expressed by `P2`.

* `P1 < P2` iff `P1 <= P2 and P1 != P2`

* vice versa for `P1 >= P2` and `P1 > P2`

### CODE REVIEW

**Comment by mmirate** [comment]
That second commit there makes the new methods raise `TypeError` when ordered with other types.

File: discord/permissions.py [hunk 1]
```diff
@@ -64,16 +64,22 @@ def __hash__(self):
...
+        else:
+            raise TypeError("cannot compare {} with {}".format(self.__class__.__name__, other.__class__name))
 
     def is_strict_subset(self, other):
-        return isinstance(other, Permissions) and self.is_subset(other) and self != other
+        return self.is_subset(other) and self != other
 
     def is_strict_superset(self, other):
-        return isinstance(other, Permissions) and self.is_subset(other) and self != other
+        return self.is_subset(other) and self != other
```
Comment (Rapptz) [inline discord/permissions.py]: Shouldn't this be `self.is_strictsuperset`?

File: discord/permissions.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ class Permissions:
         permissions via the properties rather than using this raw value.
     """
 
-    __slots__ = [ 'value' ]
+    __slots__ = ('value',)
```
Comment (Rapptz) [inline discord/permissions.py]: Why touch this?

File: discord/permissions.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ class Permissions:
         permissions via the properties rather than using this raw value.
```
Comment (Rapptz) [inline discord/permissions.py]: Documentation changes are missing.

File: discord/permissions.py (hunk 2 above)
Comment (mmirate) [inline discord/permissions.py]: Whoops, that was on accident.

**Comment by mmirate** [comment]
Okay, `__gt__` now points where it should, docstrings are on the named methods, the operators are in the table, and the spurious change is reverted.

File: discord/permissions.py (hunk 3 above)
Comment (mmirate) [inline discord/permissions.py]: Documentation is changed as of https://github.com/Rapptz/discord.py/pull/193/commits/a5c0924301233fdfe2fbb8abb0fe66dd28b98a8b

File: discord/permissions.py [hunk 4]
```diff
@@ -63,6 +75,33 @@ def __ne__(self, other):
...
+        else:
+            raise TypeError("cannot compare {} with {}".format(self.__class__.__name__, other.__class__name))
+
+    def is_strict_subset(self, other):
+        """Returns True if the permissions on other are a strict subset of those on self."""
+        return self.is_subset(other) and self != other
+
+    def is_strict_superset(self, other):
+        """Returns True if the permissions on other are a strict superset of those on self."""
+        return self.is_subset(other) and self != other
```
Comment (Rapptz) [inline discord/permissions.py]: This should be `self.is_superset` no?

Comment (mmirate) [inline discord/permissions.py]: Argh. Yes. Copy-paste typo. Fixing this momentarily.

Comment (mmirate) [inline discord/permissions.py]: And fixed.

**Comment by Rapptz** [comment]
Rebased -> https://github.com/Rapptz/discord.py/commit/21c88cf727719f954a1b31dbae5b29e4bf61e4c8

**Pull request closed without merging**